src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,142 @@
+package duke;
+
+import duke.task.Task;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import java.io.*;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+class Storage {
+
+    private String filePath;
+
+    /**
+     * Constructs a Storage object with a custom file path.
+     *
+     * @param filePath file path of the save file
+     */
+    Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Constructs a Storage object with a default file path.
+     *
+     */
+    Storage() {
+        this.filePath = getSaveFilePath();
+    }
+
+    /**
+     * Get the default file path specific to platform.
+     * @return default file path specific to platform
+     */
+    private String getSaveFilePath() {
+        URL url = getClass().getResource("/storage/data.json");
+        return url.getPath();
+        /*
+        if (System.getProperty("os.name").equals("Windows 10")) {
+            return "/Users/uicfa/Downloads/data.json";
+        } else {
+            return "/Users/leo/Downloads/data.json";
+        }
+         */
+    }
+
+    /**
+     * Creates the save file (used when it doesn't exist).
+     * @throws IOException
+     */
+    private void createSaveFile() throws IOException {
+        File file = new File(this.filePath);
+        if (!file.createNewFile()) {
+            throw new IOException();
+        }
+        writeToSaveFile("{\"data\":[]}");
+    }
+
+    /**
+     * Read and parse the save file.
+     * @return A parsed JSONObject from the save file
+     * @throws IOException
+     * @throws JSONException
+     */
+    private JSONObject readSaveFile() throws IOException, JSONException {
+        // TODO: handle the exception where data.json doesn't exist or format is wrong
+        InputStream is;
+        try {
+            is = new FileInputStream(this.filePath);
+        } catch (FileNotFoundException e) {
+            createSaveFile();
+            is = new FileInputStream(this.filePath);
+        }
+        BufferedReader buf = new BufferedReader(new InputStreamReader(is));
+        String line = buf.readLine();
+        StringBuilder sb = new StringBuilder();
+        while (line != null) {
+            sb.append(line).append("\n");
+            line = buf.readLine();
+        }
+        return new JSONObject(new JSONTokener(sb.toString()));
```

Comment:
Great choice of using JSONObject to store stuff instead of text. Makes it easier to parse and format.